lmms_eval/models/chat/async_openai.py

```
@@ -61,15 +68,36 @@ def __init__(
+        self.max_pixels = max_pixels
+        self.min_pixels = min_pixels
         if mcp_server_path is not None:
             self.mcp_client = MCPClient(mcp_server_path)
             os.makedirs(self.work_dir, exist_ok=True)
         else:
             self.mcp_client = None
 
+        accelerator = Accelerator()
+        # assert self.batch_size_per_gpu == 1, "Llava currently does not support batched generation. See https://github.com/haotian-liu/LLaVA/issues/754. HF Llava also has this issue."
+        if accelerator.num_processes > 1:
+            assert accelerator.distributed_type in [DistributedType.FSDP, DistributedType.MULTI_GPU, DistributedType.DEEPSPEED], "Unsupported distributed type provided. Only DDP and FSDP are supported."
```

Ruff (0.12.2)

Use of `assert` detected

(S101)